packages/babel-helpers/src/helpers/applyDecs2311.ts

```diff
@@ -418,49 +422,37 @@ export default /* @no-mangle */ function applyDecs2311(
           )
         ) {
           if (isField) {
-            init.push(newValue);
+            init.unshift(newValue);
           } else {
             desc[key] = newValue;
           }
         }
       }
-
-      decoratorFinishedRef.v = true;
     }
 
     if (isField || isAccessor) {
       ret.push(
         // init
-        function (instance: any, value: any) {
-          for (var i = init.length - 1; i >= 0; i--) {
-            value = init[i].call(instance, value);
-          }
-          return value;
-        },
+        runInitializers.bind(null, init, 1),
         // init_extra
-        runInitializers.bind(null, initializers),
+        runInitializers.bind(null, initializers, 0),
       );
     }
 
     if (!isField && !isClass) {
       if (isPrivate) {
         if (isAccessor) {
           // get and set should be returned before init_extra
-          ret.splice(
-            -1,
-            0,
-            _bindPropCall(desc, "get"),
-            _bindPropCall(desc, "set"),
-          );
+          ret.splice(-1, 0, _bindPropCall("get"), _bindPropCall("set"));
         } else {
           ret.push(
             kind === PROP_KIND.METHOD
               ? desc[key]
-              : _bindPropCall.call.bind(desc[key]),
+              : runInitializers.call.bind(desc[key]),
```

Comment:
```suggestion
              : Function.call.bind(desc[key]),
```

Here it doesn't actually matter that we are using `runInitializers`, because we are replacing `.call`'s function with `desc[key]` anyway. Let's just use `Function.call.bind` to avoid potential confusion.

Reply:
This will add some extra size, can we add some comments instead? :)

Reply:
Sure!